CRM/Core/ManagedEntities.php

```diff
@@ -236,6 +236,14 @@ public function reconcileUnknownModules() {
    *   Entity specification (per hook_civicrm_managedEntities).
    */
   public function insertNewEntity($todo) {
+    // If you have an existing entity and you want to convert it to managed you have to check if it already exists.
+    // Otherwise it will crash with DB error duplicate entry (eg. for PaymentProcessorType)
+    $existingEntity = civicrm_api3($todo['entity'], 'get', $todo['params']);
+    if (!empty($existingEntity['id'])) {
+      \Civi::log()->warning('Found existing entity, overwriting with managed version. Detail: ' . print_r($todo['params'], TRUE));
```

Comment:
I don't have a comment on the PR itself just as a [PSR3 thing](https://www.php-fig.org/psr/psr-3/#12-message) I'd suggest `\Civi::log()->warning('Found existing entity, overwriting with managed version. Detail: {todoparams}', ['todoparams' => $todo['params']]);` and the default PSR3 logger in civi currently does a print_r, but the placeholder allows other implementations to do whatever (json, ROT-13, ...)

Reply:
Thanks @demeritcowboy I've updated. However, I didn't add the {todoparams} because I don't see anywhere that does "token" substitution, only appending the print_r output.

Reply:
While the civi default PSR3 logger doesn't physically replace it, another implementation might. Without the placeholder an implementation doesn't know there's anything to do. The $context parameter can contain other things besides placeholders and a generic implementation can't assume they are all elements to print out. You can also chain implementations so they all might do different things.